apps/emqx/src/emqx_persistent_message.erl

```diff
@@ -52,31 +52,17 @@ is_persistence_enabled() ->
 
 -spec storage_backend() -> emqx_ds:create_db_opts().
 storage_backend() ->
-    storage_backend(emqx_config:get([session_persistence, storage])).
+    storage_backend([durable_storage, messages]).
 
 %% Dev-only option: force all messages to go through
 %% `emqx_persistent_session_ds':
 -spec force_ds() -> boolean().
 force_ds() ->
     emqx_config:get([session_persistence, force_persistence]).
 
-storage_backend(#{
-    builtin := #{
-        enable := true,
-        n_shards := NShards,
-        replication_factor := ReplicationFactor
-    }
-}) ->
-    #{
-        backend => builtin,
-        storage => {emqx_ds_storage_bitfield_lts, #{}},
-        n_shards => NShards,
-        replication_factor => ReplicationFactor
-    };
-storage_backend(#{
-    fdb := #{enable := true} = FDBConfig
-}) ->
-    FDBConfig#{backend => fdb}.
+storage_backend(Path) ->
+    ConfigTree = #{'_config_handler' := {Module, Function}} = emqx_config:get(Path),
```

Comment:
IMO odd to have that as part of the schema: it can be changed, it can end up in the `cluster.hocon` even if it does not differ from the default. Isn't it easier to assume for now that we should just call `emqx_ds_schema:something(ConfigTree)` here?

Reply:
Agree, admittedly, it's not the nicest of the hacks. However, `emqx_ds_schema:something` would have to know how to handle configuration of backends injected on EE. 

True, we could add some sort of advanced backend registration mechanism in the DS domain, but it wouldn't solve the problem of configuration translation, which is part of the emqx domain. This could be redesigned in the future.